src/sdk/pynni/nni/package_utils.py

```diff
@@ -321,12 +298,21 @@ def _generate_installation_path(sitepackages_path):
         # If system-wide python is used, we will give priority to using `local sitepackage`--"usersitepackages()" given
         # that nni exists there
         if python_sitepackage.startswith('/usr') or python_sitepackage.startswith('/Library'):
-            python_dir = try_installation_path_sequentially(site.getusersitepackages(), site.getsitepackages()[0])
+            python_dir = _try_installation_path_sequentially(site.getusersitepackages(), *site.getsitepackages())
         else:
-            python_dir = try_installation_path_sequentially(site.getsitepackages()[0], site.getusersitepackages())
-
+            python_dir = _try_installation_path_sequentially(*site.getsitepackages(), site.getusersitepackages())
     return python_dir
 
+def _try_installation_path_sequentially(*sitepackages):
+    '''Try different installation path sequentially util nni is found.
+    Return None if nothing is found
+    '''
+    for sitepackage in sitepackages:
+        for path in [Path(sitepackage).parents[2], Path(sitepackage)]:
+            if (path / 'nni' / 'main.js').is_file():
```

Comment:
what is value of `path / 'nni' / 'main.js'`?

Reply:
A `Path` object referring to `main.js` file in nni folder in `path`.